src/components/reusable/progressBar/progressBar.ts

```diff
@@ -0,0 +1,273 @@
+  ) {
+    const isIndeterminate = this.status === ProgressStatus.INDETERMINATE;
+    const resolvedValue = [
+      ProgressStatus.SUCCESS,
+      ProgressStatus.ERROR,
+    ].includes(currentStatus as ProgressStatus)
+      ? this.max
+      : currentValue;
+
+    return html`<progress
+      class="${this.getProgressBarClasses(currentStatus)}"
+      max=${this.max}
+      value=${ifDefined(
+        !isIndeterminate && resolvedValue != null ? resolvedValue : undefined
+      )}
+      aria-valuenow=${ifDefined(
+        !isIndeterminate && resolvedValue != null ? resolvedValue : undefined
+      )}
+      aria-valuemin="0"
+      aria-valuemax=${this.max}
+      aria-label=${this.label}
+    ></progress>`;
+  }
+
+  private renderProgressBarLabel(currentStatus: ProgressStatus) {
+    return html`<div class="progress-bar__upper-container">
+      <h2 class="progress-bar__label">
+        <span>${this.label}</span>
+        ${this.informationalTooltipText
+          ? html`<kyn-tooltip
+              ><span slot="anchor"
+                ><kd-icon .icon=${informationIcon}></kd-icon></span
+              >${this.informationalTooltipText}</kyn-tooltip
+            >`
+          : null}
+      </h2>
+      ${currentStatus !== ProgressStatus.INDETERMINATE
+        ? html`<div class=${`progress-bar__status-icon ${currentStatus}`}>
+            ${this.renderStatusIconOrLoader(currentStatus)}
+          </div>`
+        : null}
+    </div>`;
+  }
+
+  private renderStatusIconOrLoader(currentStatus: ProgressStatus) {
+    if (currentStatus !== ProgressStatus.ACTIVE) {
+      return html`<kd-icon
+        .icon=${currentStatus === ProgressStatus.SUCCESS
+          ? checkmarkIcon
+          : errorIcon}
+      ></kd-icon>`;
+    }
+
+    if (this.showInlineLoadStatus) {
+      return html`<p>
+        <span>${this._percentage}%</span
+        ><kyn-loader-inline status="active"></kyn-loader-inline>
+      </p>`;
+    }
+
+    return null;
+  }
+
+  override firstUpdated() {
+    if (this.status === ProgressStatus.ACTIVE) {
+      this.startProgress();
+    }
+  }
+
+  override updated(changedProperties: Map<string, any>) {
+    if (changedProperties.has('status') || changedProperties.has('value')) {
+      this.cancelAnimation();
+      this._running = false;
+
+      if (this.status === ProgressStatus.ACTIVE) {
+        this._running = true;
+        this.startProgress();
+      }
+    }
+  }
+
+  private getProgressBarClasses(status: string) {
+    return classMap({
+      'progress-bar__main': true,
+      [`progress-bar__${status}`]: true,
+    });
+  }
+
+  private getHelperText() {
+    if (this.helperText) {
+      return this.helperText;
+    }
+
+    if (this.status === ProgressStatus.ACTIVE) {
+      const formattedProgress =
+        this.unit === '%'
+          ? this._progress.toFixed(0)
+          : this._progress.toFixed(1);
+      const formattedMax =
+        this.unit === '%' ? this.max.toFixed(0) : this.max.toFixed(1);
```

Comment:
Might be simpler to just use the raw values provided for `value` and `max`.

Reply:
in this case, `_progress` is used internally for the smooth animations, incrementing the bar. while `value` could be just a static, final numeric value.

if it's alright, i'll just strip the formatting and use the raw `this._progress` and `this.max` values